### PR TITLE
Enable quantization for set of public models

### PR DESCRIPTION
**Pull request by Anna-Mironova**
to extend number of quantized public models extend description of datasets with
[PASCAL-S](http://cbs.ic.gatech.edu/salobj/)
[CoNLL2003 Named Entity Recognition](https://www.aclweb.org/anthology/W03-0419/)
[MRL Eye](http://mrl.cs.vsb.cz/eyedataset)
[Labeled Faces in the Wild (LFW)](http://vis-www.cs.umass.edu/lfw/)
[NYU Depth Dataset V2](https://cs.nyu.edu/~silberman/datasets/nyu_depth_v2.html)
and following models marked as quantizable
Sphereface
bert-base-ner
deeplabv3
dla-34
f3net
faceboxes-pytorch
faster_rcnn_inception_v2_coco
fcrn-dp-nyu-depth-v2-tf
googlenet-v2-tf
hbonet-0.5
hbonet-1.0
mobilefacedet-v1-mxnet
mobilenet-ssd
nfnet-f0
open-closed-eye-0001
regnetx-3.2gf
retinanet-tf
shufflenet-v2-x1.0
ssd300
ssd512
ssd_resnet50_v1_fpn_coco
yolo-v1-tiny-tf
yolo-v2-tf
yolo-v3-tiny-tf
yolo-v4-tf

### CODE REVIEW

**Review by vladimir-dudnik — APPROVED** [review]

File: data/datasets.md [hunk 1]
```diff
@@ -141,3 +141,111 @@ To use this dataset with OMZ tools, make sure `<DATASET_DIR>` contains the follo
...
+### Datasets in dataset_definitions.yml
+* `PASCAL-S` used for evaluation models on PASCAL-S dataset for salient object detection task. (model examples: [`f3net`](../models/public/f3net/README.md))
+
+## [CoNLL2003 Named Entity Recognition](https://www.aclweb.org/anthology/W03-0419/)
+
+### How download dataset
+
+To download CoNLL2003 dataset, you need to follow the steps below:
+1. Go to the [CoNLL 2003 (English) Dataset](https://deepai.org/dataset/conll-2003-english) page in the [DeepAI Datasets](https://deepai.org/datasets) website
+2. Select [`DOWNLOAD`](https://data.deepai.org/conll2003.zip) and download archive
```
Comment (eaidova) [inline data/datasets.md:172]: Are you tried to convert it?
I used another source for downloading data, so possibly some issues in convertor can be

File: data/datasets.md [hunk 2]
```diff
@@ -141,3 +141,111 @@ To use this dataset with OMZ tools, make sure `<DATASET_DIR>` contains the follo
...
+3. Unpack archive
+
+### Files layout
+
+To use this dataset with OMZ tools, make sure `<DATASET_DIR>` contains the following:
+
+* `CONLL-2003` - directory containing annotation files
+    * `valid.txt` - annotation file for CoNLL2003 development set
```
Comment (eaidova) [inline data/datasets.md:180]:
```suggestion
    * `valid.txt` - annotation file for CoNLL2003 validation set
```

File: data/datasets.md [hunk 3]
```diff
@@ -141,3 +141,111 @@ To use this dataset with OMZ tools, make sure `<DATASET_DIR>` contains the follo
...
+
+To use this dataset with OMZ tools, make sure `<DATASET_DIR>` contains the following:
+
+* `nyudepthv2/val` - directory with dataset official data and converted images and depth map
+    * `official` - directory with data stored in original hdf5 format
+    * `converted` - directory with converted data
+        * `images` - directory with converted images
+        * `depth` -  directory with depth maps
+
+Note: If dataset is used in the first time, please set `allow_convert_data: True` in annotation conversion parameters for this dataset in `dataset_definitions.yml` to get converted data.
```
Comment (eaidova) [inline data/datasets.md:248]: possibly command line how to convert without running AC or quantization should be added (via `convert_annotation`)

File: tools/accuracy_checker/accuracy_checker/annotation_converters/nyu_depth.py [hunk 4]
```diff
@@ -28,33 +28,44 @@ def parameters(cls):
...
-        if self.data_dir is None:
-            self.images_dir = get_path(self.images_dir, is_directory=True)
-            self.depths_dir = get_path(self.depths_dir, is_directory=True)
-        else:
+
+        if self.allow_convert_data:
+            if self.data_dir is None:
+                raise ConfigError('please provide data_dir to convert data from hdf5 format')
+            if isinstance(h5py, UnsupportedPackage):
+                h5py.raise_error(self.__provider__)
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/annotation_converters/nyu_depth.py:48]: logically, even user specify data dir, but does not have installed dependencies, he will not be able to convert, so h5py should be checked before data_dir
```suggestion
           if isinstance(h5py, UnsupportedPackage):
                h5py.raise_error(self.__provider__)
            if self.data_dir is None:
                raise ConfigError('please provide data_dir to convert data from hdf5 format')
```

Comment (eaidova) [reply tools/accuracy_checker/accuracy_checker/annotation_converters/nyu_depth.py:48]: BTW, now specifying all 3 paths looks reductant, possibly this moment also should be refactored

File: models/public/mobilefacedet-v1-mxnet/quantization.yml [hunk 5]
```diff
@@ -0,0 +1,10 @@
+compression:
+  algorithms:
+    - name: DefaultQuantization
+      params:
+        num_samples_for_tuning: 2000
+        preset: accuracy
+        stat_subset_size: 300
+        use_fast_bias: False
+        use_layerwise_tuning: False
+  dump_intermediate_model: True
```
Comment (IRDonch) [inline models/public/mobilefacedet-v1-mxnet/quantization.yml:10]: Is this option needed? I don't really know what it does, but from the name it doesn't seem like it would affect the quantization result.

**Review by IRDonch — CHANGES_REQUESTED** [review]

**Comment by IRDonch** [comment]
@Anna-Mironova It would be nice to add a model with a custom quantization config to the list of representative models for testing the quantizer. Could you check if any of those models takes a reasonable time to get quantized (a few minutes) and if so, add it to `tools/downloader/tests/representative-models-quantization.lst`?

File: data/datasets.md (hunk 2 above)
Comment (Anna-Mironova) [reply data/datasets.md]: Done

File: data/datasets.md (hunk 3 above)
Comment (Anna-Mironova) [reply data/datasets.md]: Done

File: models/public/mobilefacedet-v1-mxnet/quantization.yml (hunk 5 above)
Comment (Anna-Mironova) [reply models/public/mobilefacedet-v1-mxnet/quantization.yml]: Removed

File: data/datasets.md [hunk 6]
```diff
@@ -141,3 +141,110 @@ To use this dataset with OMZ tools, make sure `<DATASET_DIR>` contains the follo
...
+* `nyudepthv2/val` - directory with dataset official data and converted images and depth map
+    * `official` - directory with data stored in original hdf5 format
+    * `converted` - directory with converted data
+        * `images` - directory with converted images
+        * `depth` -  directory with depth maps
+
+Note: If dataset is used in the first time, please set `allow_convert_data: True` in annotation conversion parameters for this dataset in `dataset_definitions.yml`  or use [convert.py](../tools/accuracy_checker/accuracy_checker/annotation_converters/convert.py) and  following command line to get converted data .
+
+```sh
+OMZ_ROOT/accuracy_checker/accuracy_checker/annotation_converters/convert.py  nyu_depth_v2 --data_dir <DATASET_DIR>/nyudepthv2/val/official --allow_convert_data True
```
Comment (eaidova) [inline data/datasets.md:246]: there is entry point for conversion (like `accuracy_check`)
```suggestion
convert_annotation  nyu_depth_v2 --data_dir <DATASET_DIR>/nyudepthv2/val/official --allow_convert_data True
```

File: tools/accuracy_checker/accuracy_checker/annotation_converters/nyu_depth.py (hunk 4 above)
Comment (Anna-Mironova) [reply tools/accuracy_checker/accuracy_checker/annotation_converters/nyu_depth.py]: Done

File: data/datasets.md [hunk 7]
```diff
@@ -141,3 +141,110 @@ To use this dataset with OMZ tools, make sure `<DATASET_DIR>` contains the follo
 ### Datasets in dataset_definitions.yml
 * `VOC2007_detection` used for evaluation models on VOC2007 dataset for object detection task. Background label + label map with 20 object categories are used. (model examples: [`mobilenet-ssd`](../models/public/mobilenet-ssd/README.md), [`ssd300`](../models/public/ssd300/README.md))
 * `VOC2007_detection_no_bkgr` used for evaluation models on VOC2007 dataset for object detection tasks. Label map with 20 object categories is used.(model examples: [`yolo-v1-tiny-tf`](../models/public/yolo-v1-tiny-tf/README.md))
+
+## [PASCAL-S](http://cbs.ic.gatech.edu/salobj/)
+
+### How download dataset
+
+To download PASCAL-S dataset, you need to follow the steps below:
+1. [Download](http://cbs.ic.gatech.edu/salobj/download/salObj.zip) archive from the [The Secrets of Salient Object Segmentation](http://cbs.ic.gatech.edu/salobj/) website
```
[the remark this answers is not in the document]
Comment (Anna-Mironova) [reply data/datasets.md]: Done

File: data/datasets.md (hunk 6 above)
Comment (Anna-Mironova) [reply data/datasets.md]: Done

File: tools/downloader/tests/representative-models-quantization.lst [hunk 8]
```diff
@@ -1,3 +1,4 @@
 # A list of models for testing quantization features.
 
 mobilenet-v1-0.25-128
+hbonet-0.5
```
Comment (IRDonch) [inline tools/downloader/tests/representative-models-quantization.lst:4]: Please mark these to make it clear why they are here, e.g.:

```
mobilenet-v1-0.25-128 # default config
hbonet-0.5 # custom config
```

Comment (Anna-Mironova) [reply tools/downloader/tests/representative-models-quantization.lst]: Done

**Review by eaidova — APPROVED** [review]

File: models/public/ssh-mxnet/model.yml [hunk 9]
```diff
@@ -32,4 +32,5 @@ model_optimizer_args:
   - --mean_values=data[123.68,116.779,103.939]
   - --reverse_input_channels
 framework: mxnet
+quantizable: yes
```
Comment (IRDonch) [inline models/public/ssh-mxnet/model.yml:35]: Given that `ssh-mxnet` is (at least temporarily) unavailable, I suggest you revert this for now, so that the PR can be properly tested. If it comes back, you can make it quantizable in a separate PR.

**Review by IRDonch — APPROVED** [review]
